src/WebJobs.Script/Utility.cs

```diff
@@ -879,6 +881,31 @@ public static bool IsResourceAzureBlobWithoutSas(Uri resourceUri)
             return true;
         }
 
+        public static string GetSasTokenExpirationDate(Uri resourceUri)
+        {
+            // Screen out URLs with an SAS token
+            var queryParams = HttpUtility.ParseQueryString(resourceUri.Query.ToLower());
+            if (queryParams != null && !string.IsNullOrEmpty(queryParams[SasTokenExpirationDate]))
+            {
+                return queryParams[SasTokenExpirationDate];
+            }
+
+            return string.Empty;
+        }
+
+        public static string GetSasTokenExpirationDateFromSasSignature(string azureWebJobsStorage)
+        {
+            var splitStorage = azureWebJobsStorage.Split(';');
+            var sasToken = splitStorage[splitStorage.Length - 1];
+            var queryParams = HttpUtility.ParseQueryString(sasToken);
+            if (queryParams != null && !string.IsNullOrEmpty(queryParams[SasTokenExpirationDate]))
```

Comment:
nit: I think the null check can be removed as the [method does not return null](https://learn.microsoft.com/en-us/dotnet/api/System.Web.HttpUtility.ParseQueryString?view=netstandard-2.1) ?